### DISCUSSION TITLE
transform - undefined function replace_pattern

### DISCUSSION THREAD

**Discussion by aprvital2000** [open-telemetry/opentelemetry-collector-contrib#11853] [Q&A] [answered]
Hi,
I'm trying to use transform.replace_pattern, transform.replace_all_patterns. I see following error. Using splunk-otel-collector-0.54.0

Error: failed to get config: invalid configuration: processor "transform" has invalid configuration: undefined function replace_pattern
Error: failed to get config: invalid configuration: processor "transform" has invalid configuration: undefined function replace_all_patterns

Where as, I'm able to use replace_match, replace_all_matches.
Are the functions ready to use? or, what could be the issue?

**Comment by TylerHelmuth** [comment, marked as answer]
replace_pattern and replace_all_patterns are merged into main but have not been released yet.  They will be available in the 0.55.0 version of Collector Contrib

**Reply by aprvital2000** [reply]
Thank You for the confirmation
Is there a tentative date for 0.55.0 release?

**Reply by TylerHelmuth** [reply]
July 7th

**Reply by TylerHelmuth** [reply]
The collector releases every 2 weeks.

**Reply by kumachop2** [reply]
@TylerHelmuth, We are using below transform rule to replace phone number. Its it is working fine with regex101 but some issues with splunk client v0.61.0 or v0.63.0 process. Here are the details. Please check and advice

rule: - replace_pattern(body.message, "([\\\">])?([0-9]{3})[- ]*([0-9]{3})[- ]*([0-9]{4})([\\\"<])?", "$1**$4$5")

**Message:**
Address {            addressType: Permanent            address1: 1017 THRUSH LN            address2: null            city: AUDUBON            state: PA            zip: 19403        }]        contact: class Contact {            eMail: t-rex@example.com            phoneNumber: 2745888692        }        citizenshipStatus: class 

**Regex101 output:**
     address2: null            city: AUDUBON            state: PA            zip: 19403        }]        contact: class Contact {            eMail: t-rex@example.com            phoneNumber: **8692        }        citizenshipStatus: class 

**Splunk Log Observer output:**
addresses: [class Address {            addressType: Permanent            address1: 1017 THRUSH LN            address2: null            city: AUDUBON            state: PA            zip: 19403        }]        contact: class Contact {            eMail: t-rex@example.com            phoneNumber: **2        }        citizenshipStatus: class CitizenshipStatus {
